LoopKit/HealthKitSampleStore.swift

```diff
@@ -221,51 +187,59 @@ public class HealthKitSampleStore {
     /// - Parameters:
     ///   - query: The query which triggered the update
     ///   - error: An error during the update, if one occurred
-    internal final func observeUpdates(to query: HKObserverQuery, completionHandler: HKObserverQueryCompletionHandler?, error: Error?) {
-        self.log.default("observeUpdates() - HK Notifying of changes.")
+    internal final func observeUpdates(to query: HKObserverQuery, completionHandler: @escaping HKObserverQueryCompletionHandler, error: Error?) {
 
         if let error = error {
-            self.log.error("%@ notified with changes with error: %{public}@", query, String(describing: error))
-            completionHandler?()
+            self.log.error("Observer query %@ notified of error: %{public}@", query, String(describing: error))
+            completionHandler()
             return
         }
+        
+        self.log.default("observeUpdates invoked - queueing handling")
 
-        // Hold the completion handler (calling any existing ones) until our next anchored object query returns
-        callAndUpdateObserverQueryCompletionHandler(newValue: completionHandler)
+        observerQueryUpdateHandlerQueue.async {
 
-        let queryAnchor = self.queryAnchor
-        
-        log.default("%@ notified with changes. Fetching from: %{public}@", query, queryAnchor.map(String.init(describing:)) ?? "0")
+            let queryAnchor = self.queryAnchor
+            
+            self.log.default("%@ notified with changes. Fetching from: %{public}@", query, queryAnchor.map(String.init(describing:)) ?? "0")
+            
+            let semaphore = DispatchSemaphore(value: 0)
```

Comment:
[nit] Or consider a `DispatchGroup` (now that I think about it) since that is more commonly used in Loop code.

Reply:
DispatchGroup is good for aggregate synchronization of a bunch of tasks; I want to serialize this work, so I think a queue is a better fit here.